Marlin/src/HAL/HAL_STM32F1/HAL.cpp

```diff
@@ -277,8 +277,9 @@ void HAL_adc_init(void) {
 }
 
 void HAL_adc_start_conversion(const uint8_t adc_pin) {
-  TEMP_PINS pin_index = ADC_PIN_COUNT;
+  TEMP_PINS pin_index;
   switch (adc_pin) {
+    default: return;
```

Comment:
seems ok for the compiler, i would put it at the end for the switch... but its more a code style thing